pydrive2/auth.py

```diff
@@ -367,25 +328,55 @@ def LoadCredentialsFile(self, credentials_file=None):
         :raises: InvalidConfigError, InvalidCredentialsError
         """
         if credentials_file is None:
-            self._default_storage = self._storages["file"]
-            if self._default_storage is None:
+            credentials_file = self.settings.get("save_credentials_file")
+            if credentials_file is None:
                 raise InvalidConfigError(
-                    "Backend `file` is not configured, specify "
-                    "credentials file to read in the settings "
-                    "file or pass an explicit value"
+                    "Please specify credentials file to read"
                 )
-        else:
-            self._default_storage = Storage(credentials_file)
 
         try:
-            self.credentials = self._default_storage.get()
+            auth_info = self.storage.read_credentials(credentials_file)
+        except FileNotFoundError:
+            # if credential found was not found, raise the error for handling
+            raise
         except OSError:
+            # catch other errors
             raise InvalidCredentialsError(
                 "Credentials file cannot be symbolic link"
             )
 
-        if self.credentials:
-            self.credentials.set_store(self._default_storage)
+        try:
+            self._credentials = google.oauth2.credentials.Credentials.from_authorized_user_info(
+                auth_info, scopes=auth_info["scopes"]
+            )
+        except ValueError:
+            # if saved credentials lack a refresh token
+            # handled for backwards compatibility
+            warn(
```

Comment:
could you please give a bit more context on this? where does this requirement come from?

Reply:
In the new [google.oauth2.Credentials](https://github.com/googleapis/google-auth-library-python/blob/main/google/oauth2/credentials.py#L51) object, the only one way to skip auth + consent & directly load the credentials object is via the `from_authorized_user_file` or `from_authorized_user_info` method. This method requires a saved credentials with a refresh token already so that auto-refresh can happen

Reply:
okay, but I don't understand in what cases we would have credentials w/o a refresh token in the previous implementation? It might I'm missing something that I'm not aware about.

Reply:
Apologies for some unnecessary checks here. I realized that the previous serializations of `Oauth2Credentials` [here](https://github.com/googleapis/oauth2client/blob/master/oauth2client/client.py#L441) would dump the entire `self.__dict__`. This would always contain the `refresh_token` key.

Shall I remove this warning as it's unnecessary?

Reply:
To clarify on what's happening
New google auth's `from_authorized_user_file` / `from_authorized_user_info` requires the saved credentials to have the `refresh_token` key. This means that:
- Refreshable credentials are considered authorized credentials & can skip the auth & consent steps
- Non-refreshable credentials are not considered authorized & cannot be loaded by the above method

**Implications on old oauth2client Credentials**
- Credentials objects can be saved when `refresh_token` = None
- `refresh_token` key  is _not stripped_ from the saved_credentials.json file
- New google auth can still load this "authorized credentials" without error. However, the unexpected outcome is that user would still have to go through the whole consent process

**Implications on new google-auth Credentials**
- Credentials objects can be saved when `refresh_token` = None
- However `refresh_token` key  _will be stripped_ from the saved_credentials.json (see [here](https://github.com/googleapis/google-auth-library-python/blob/main/google/oauth2/credentials.py#L404))
- Google auth will raise an error when loading these credentials because of missing refresh token. The temporary workaround here is to catch the error and manually create the credentials

Reply:
@shcheklein How should I approach this? My current implementation is to catch the error when loading non-refreshable credentials & just manually create them

Reply:
thanks @junpeng-jp for the clarification! You've described this from the `oauth2client` and `google-auth` libraries perspective, but was there / is there a practical scenario for the PyDrive2 GoogleAuth itself? Was there a case in which it was saving w/o refresh token?